reactor-netty-core/src/main/java/reactor/netty/resources/DefaultPooledConnectionProvider.java

```diff
@@ -377,16 +377,14 @@ public Context currentContext() {
 
 	static final class PooledConnection implements Connection, ConnectionObserver {
 		final Channel channel;
-		@SuppressWarnings("deprecation")
-		final MonoProcessor<Void> onTerminate;
+		final Sinks.Empty<Void> onTerminate;
 		final InstrumentedPool<PooledConnection> pool;
 
 		PooledRef<PooledConnection> pooledRef;
 
-		@SuppressWarnings("deprecation")
 		PooledConnection(Channel channel, InstrumentedPool<PooledConnection> pool) {
 			this.channel = channel;
-			this.onTerminate = MonoProcessor.create();
+			this.onTerminate = Sinks.empty();
```

Comment:
same here, `Sinks.unsafe().empty()`

Reply:
It is guaranteed that this will be invoked in an event loop. I'll change it to unsafe.

Reply:
done